pkg/oci/client/client.go

```diff
@@ -142,3 +162,52 @@ func newConfigurationProvider(cfg *Config) (common.ConfigurationProvider, error)
 	}
 	return conf, nil
 }
+
+func configureCustomTransport(baseClient common.BaseClient) error {
```

Comment:
Maybe make this a pointer arg, just to make it clear that the intention of this function is to modify the baseClient?